bw2data/project.py

```diff
@@ -52,13 +56,52 @@ def __lt__(self, other):
         else:
             return self.name.lower() < other.name.lower()
 
+    def _get_base_directories(self):
+        envvar = maybe_path(os.getenv("BRIGHTWAY2_DIR"))
+        if envvar:
+            if not envvar.is_dir():
+                raise OSError(
+                    (
+                        "BRIGHTWAY2_DIR variable is {}, but this is not"
+                        " a valid directory"
+                    ).format(envvar)
+                )
+            else:
+                print(
+                    "Using environment variable BRIGHTWAY2_DIR for data "
+                    "directory:\n{}".format(envvar)
+                )
+                return envvar.absolute()
+
+        return Path(PlatformDirs("Brightway3", "pylca").user_data_dir)
+
+    @property
+    def dir(self):
+        return Path(self._get_base_directories()) / safe_filename(
+            self.name, full=self.full_hash
+        )
```

Comment:
This is copied over from `ProjectManager`, we need to decide if the patches go on DB or on a file, then I'll refactor. However, I think that projects should be more encapsulated and be responsible for their own internal management, and `ProjectManager` can play a more facade-ish role.

Reply:
@bbguimaraes